src/hotspot/share/prims/unsafe.cpp

```diff
@@ -493,7 +493,11 @@ static jlong find_field_offset(jclass clazz, jstring name, TRAPS) {
   for (JavaFieldStream fs(k); !fs.done(); fs.next()) {
     Symbol *name = fs.name();
     if (name->equals(utf_name)) {
-      offset = fs.offset();
+      if (fs.access_flags().is_static()) {
+        offset = -1;
+      } else {
```

Comment:
Since `offset` is initialised to `-1`, this can simply do:
```suggestion
      if (!fs.access_flags().is_static()) {
```

Reply:
True. Meanwhile I think I will add a comment to indicate this intends to fail fast when a matching static field is encountered.